### PR TITLE
Ecr registry

### PR DESCRIPTION
**Pull request by sshookman**
[#128]

### Solution
Adds the ability for the registry component to publish the project docker image to AWS ECR by allowing the user to specify aws details in the configuration.

This requires the user to have `~/.aws/` credentials setup and the aws CLI installed.

### CODE REVIEW

**Comment by sshookman** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- skelebot/components/registry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/skelebot/skelebot/pullRequest?prid=4843416&bid=16078755)

File: docs/publishing.md [hunk 1]
```diff
@@ -37,6 +37,20 @@ If you would like to make use of any custom tags when publishing the image, the
...
----
+### AWS ECR
+Skelebot supports publishing to AWS ECR as long as you have the proper credentials setup in `~/.aws/` as well as the [aws cli](https://aws.amazon.com/cli/) installed.
+
+To allow for ECR host authentication, the `aws` section will need to be added to your config.
+
+```
+components:
+  registry:
+    host: 12345.ecr..amazonaws.com
```
Comment (jagmoreira) [inline docs/publishing.md]:
```suggestion
    host: 12345.ecr.amazonaws.com
```

File: skelebot/systems/execution/docker.py [hunk 2]
```diff
@@ -103,6 +124,10 @@ def push(config, host=None, port=None, user=None, tags=None):
     tags = [] if tags is None else tags
     tags = tags + [config.version, "latest"]
 
+    status = 0
     for tag in tags:
-        execute(TAG_CMD.format(src=imageName, image=image, tag=tag))
-        execute(PUSH_CMD.format(image=image, tag=tag))
+        status = execute(TAG_CMD.format(src=imageName, image=image, tag=tag))
+        status = execute(PUSH_CMD.format(image=image, tag=tag))
```
Comment (jagmoreira) [inline skelebot/systems/execution/docker.py]: It may be a good idea to add a check for a failed push here

Comment (sshookman) [reply skelebot/systems/execution/docker.py]: The execute function itself raises an exception if it fails. returning the status is actually kinda redundant... I would like to totally revisist error handling in v2.0.

**Comment by sshookman** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- skelebot/components/registry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/skelebot/skelebot/pullRequest?prid=4843416&bid=16078755)

**Pull request merged**